### PR TITLE
Solution to Week2 Algorithm

### PR DESCRIPTION
**Pull request by AdefabiAdekunle**

### CODE REVIEW

File: src/binary-reversal/index.js [hunk 1]
```diff
@@ -3,6 +3,29 @@
...
+        .join(""),
+      2
+    ));
+  }
+
+  let lengthOfZeroToAdd = 8 - (lengthOfBinaryValue % 8);
+
+  let reverseBinaryValue = binaryValue.split("").reverse().join("");
+
+  let stringZeroToAdd = "0".repeat(lengthOfZeroToAdd);
```
Comment (bondz) [inline src/binary-reversal/index.js:24]: This is essentially `''.padStart(8, 0)`

File: src/list-sorting/index.js [hunk 2]
```diff
@@ -1,3 +1,45 @@
...
+      colIndexOfNeedle[colIndexOfNeedle.length - 1],
+    ];
+  }
+  if (
+    rowIndexOfNeedle.length === 0 &&
+    colIndexOfNeedle.length === 0 &&
+    indexOfNeedle.length > 0
+  ) {
+    return indexOfNeedle[indexOfNeedle.length - 1];
+  }
```
Comment (bondz) [inline src/list-sorting/index.js:40]: Too many checks and magically removing `-1` doesn't help. Can be simplified.

File: src/binary-reversal/index.js (hunk 1 above)
Comment (AdefabiAdekunle) [reply src/binary-reversal/index.js:24]: Thanks @bondz , i never knew there is a syntax like that

File: src/list-sorting/index.js (hunk 2 above)
Comment (AdefabiAdekunle) [reply src/list-sorting/index.js:40]: Noted @bondz

**Pull request merged**